### PR TITLE
Feature/hmac authentication

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
         "nelmio/api-doc-bundle": "^2.9",
         "friendsofsymfony/user-bundle": "^1.3",
         "jms/serializer-bundle": "^1.0",
-        "TransformCore/hay-local-health-authority-bundle": "^1.0.0"
+        "TransformCore/hay-local-health-authority-bundle": "^1.0.0",
+        "acquia/http-hmac-php": "^1.0"
```
Comment (eddiejaoude) [inline composer.json]: Is the hmac library well supported?

Comment (eddiejaoude) [inline composer.json]: Only 5 stars & 5 forks

Comment (brettminnie) [inline composer.json]: It's the most well supported I found

Comment (brettminnie) [inline composer.json]: Hrmm, was well layer out and easily extended, the bundle was quite
intrusive

I'll take a second look, but I got it working as expected

Comment (brettminnie) [inline composer.json]: Based on results from packagist it was one of the most widely used, the other contender did not really suit the purposes here

Comment (eddiejaoude) [inline composer.json]: Fair enough :+1:

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Listener/RequestListener.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    /**
+     * @param GetResponseEvent $event
+     *
+     * Attempts to authenticate the user via hmac
+     *
+     * @throws MalformedRequestException
+     * @throws TimestampOutOfRangeException
+     * @throws KeyNotFoundException
+     * @throws InvalidSignatureException
+     * @throws \Exception
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Listener/RequestListener.php]: Is this correct? Where is the **Exception** Thrown?

Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Listener/RequestListener.php]: Ignore that, I just read further down :smile:

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Service/DrinkingService.php [hunk 3]
```diff
@@ -17,12 +17,13 @@ class DrinkingService extends RestfulService
     const ENTITY_NAME = 'TransformCore\PHE\HayApi\PersistenceBundle\Entity\Questions\Drinking';
 
     /**
-     * @param int          $id
-     * @param ParamFetcher $paramFetcher
-     * @return View
+     * @param $id
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Service/DrinkingService.php]: `id` type?

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Service/EatingService.php [hunk 4]
```diff
@@ -15,12 +16,13 @@ class EatingService extends RestfulService
     const ENTITY_NAME = 'TransformCore\PHE\HayApi\PersistenceBundle\Entity\Questions\Eating';
 
     /**
-     * @param int          $id
-     * @param ParamFetcher $paramFetcher
-     * @return View
+     * @param $id
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Service/EatingService.php]: `id` type?

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Service/RestfulService.php [hunk 5]
```diff
@@ -19,6 +19,8 @@
 
     const STATUS_ENTITY_NOT_FOUND   = 404;
 
+    const STATUS_NOT_AUTHENTICATED  = 401;
+
     const STATUS_INTERNAL_ERROR     = 500;
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Service/RestfulService.php]: You could use existing **status codes** in Symfony **Response** object

Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Service/RestfulService.php]: Good spot

**Pull request merged**